### PR TITLE
Add a flag to "flutter build apk" that can package additional files i…

### PR DESCRIPTION
**Pull request by jason-simmons**
…nto the APK

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/build_apk.dart [hunk 1]
```diff
@@ -155,6 +156,9 @@ class BuildApkCommand extends FlutterCommand {
     argParser.addOption('flx',
       abbr: 'f',
       help: 'Path to the FLX file. If this is not provided, an FLX will be built.');
+    argParser.addOption('add-file',
+      help: 'Add a file to the APK.  Must have the format [path in APK]=[local file path]',
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: Perhaps: `Add a file to the APK (must have the format <path/in/APK>=<local/file/path>).`.

Whatever text you think works best for the cli option; we should use a full sentence through, to match the other options.

File: packages/flutter_tools/lib/src/commands/build_apk.dart [hunk 2]
```diff
@@ -295,6 +313,10 @@ int _buildApk(
     String abiDir = platform == TargetPlatform.android_arm ? 'armeabi-v7a' : 'x86_64';
     artifactBuilder.add(components.libSkyShell, 'lib/$abiDir/libsky_shell.so');
 
+    for (String relativePath in components.extraFiles.keys) {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: braces not needed here

**Comment by devoncarew** [comment]
lgtm!

Do you expect this to be something rare for people to do, or more frequent? I ask because we might want to instead have something like an `--include-extras` flag, which points to a declarative file on disk that specifies which additional files to include. That way you don't have to re-specify the files each time you run `flutter build apk`.

Or we could do something like have a section in flutter.yaml which listed additional files for the APK, or have a well known dir inside of `android/` that we always copy in. I don't have a good sense of how this'll be used -

File: packages/flutter_tools/lib/src/commands/build_apk.dart (hunk 1 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: done

File: packages/flutter_tools/lib/src/commands/build_apk.dart (hunk 2 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/commands/build_apk.dart]: done

**Comment by jason-simmons** [comment]
Currently this is intended for use in development of features like AOT compilation.

I don't think it will be common for developers to add custom files to APKs created with the flutter tool.  I suspect that many applications that could need this will move to packaging APKs with another tool such as Gradle.

**Pull request merged**